lib/rewrite/tests/test_rewrite.c

```diff
@@ -97,277 +95,263 @@ rewrite_teardown(LogMessage *msg)
   cfg_free(configuration);
 }
 
-void
-test_condition_success(void)
+#undef assert_msg_field_equals
+#define assert_msg_field_equals(msg, name, expected, expected_value_len) \
+  do { \
+   gssize value_len = -1; \
+   const gchar *actual = log_msg_get_value_by_name(msg, name, &value_len); \
+   \
+   gssize exp_value_len = expected_value_len; \
+   if (exp_value_len < 0) \
+     exp_value_len = strlen(expected); \
+   \
+   cr_assert_eq(value_len, exp_value_len, "Message name(%s) value_len(%ld) expected_value_len(%ld) at line %d", name, value_len, exp_value_len, __LINE__); \
+   cr_assert_str_eq(actual, expected, "Message name(%s) actual(%s) expected(%s) at line %d", name, actual, expected, __LINE__); \
```

Comment:
Is `__LINE__` required here? Without it, this macro can be refactored into an assert function.

Reply:
Without `__LINE__`, refactored into a separate function, it looks like something like this:

```
[----] ../lib/rewrite/tests/test_rewrite.c:123: Assertion failed: Message device_id(00100) value_len(0) -1(5)
[FAIL] rewrite::condition_success: (0,01s)
```

The line number here points to the assert function, not to its caller, but the call side can be found easily based on the actual parameters.

What do you think?

Reply:
That's fine by me. Before criterion it was painful.